src/main/java/com/google/api/codegen/discogapic/transformer/DiscoGapicParser.java

```diff
@@ -100,7 +101,24 @@ public static String getCanonicalPath(Method method) {
       }
       namePattern = m.appendTail(sb).toString();
     }
-    return namePattern;
+
+    // Assume, based on the Google Compute API, that there is no more than one wildcard segment in a row.
+    // Now the path segments alternate between exactly one literal segment and exactly one wildcard segment.
```

Comment:
Does this always have to be the case? What about something like `projects/{project}/qualifier/regions/{region}` - is this possible? And if so, do we also need to handle potential collisions, such as `projects/{project}/alternate/regions/{region}`?

Reply:
Definitely valid points, thanks!

- Removed the assumption that "there is no more than one wildcard segment in a row".
- Yes, now handling potential conflicts like the ones you gave as examples. Those would now be two different resource names, with the java class names `ProjectQualifierRegionName` and `ProjectAlternateRegionName` respectively.